R/supervised-model-deployment.R

```diff
@@ -302,6 +349,36 @@ SupervisedModelDeployment <- R6Class("SupervisedModelDeployment",
 
     #Deploy the Model
     deploy = function() {
+    },
+    
+    # A function to get the ordered list of top factors with parameters to 
+    # choose how many factors to include and whether or not to include weights
+    getTopFactors = function(numberOfFactors = NA, includeWeights = FALSE) {
+      # Include all factors by default
+      if (is.na(numberOfFactors)) {
+        numberOfFactors <- ncol(private$orderedFactors)
+      }
+      # Don't include more factors than exist
+      numberOfFactors <- min(numberOfFactors, ncol(private$orderedFactors))
+      # Include grain column
+      topFactorsDf <- data.frame(id = private$grainTest)
+      if (includeWeights) {
+      # Get factor weights
+        factorWeights <- t(sapply(1:nrow(private$multiplyRes),
+                                  function(i)
+                                        private$multiplyRes[i, ][order(private$multiplyRes[i, ],
+                                                                  decreasing = TRUE)]))
+      }
+      # Add each of the top factors
+      for (i in 1:numberOfFactors) {
+        ithTopFactor <- paste0("Factor", i, "TXT")
+        topFactorsDf[[ithTopFactor]] <- private$orderedFactors[, i]
+        if (includeWeights) {
+          ithWeight <- paste0("Factor", i, "Weight")
+          topFactorsDf[[ithWeight]] <- as.numeric(factorWeights[, i])
+        }
+      }
+      return(topFactorsDf)
```

Comment:
It doesn't clean up the code in supervised model deployment very much, but it does mean we can get rid of one of the unit tests (making sure getTopFactors and getOutDf list the same top factors)